File: app/src/main/java/com/example/gym_workout/DatabaseHelper.java
    package com.example.gym_workout;


    import android.content.Context;
    import android.database.Cursor;
    import android.database.sqlite.SQLiteDatabase;
    import android.database.sqlite.SQLiteOpenHelper;

    public class DatabaseHelper extends SQLiteOpenHelper {

        private static final String DATABASE_NAME = "gym.db";
        private static final int DATABASE_VERSION = 2;

        public static final String TABLE_USERS = "users";
        public static final String COLUMN_ID = "id";
        public static final String COLUMN_USERNAME = "username";
        public static final String COLUMN_PASSWORD = "password";

        private static final String TABLE_CREATE =
                "CREATE TABLE " + TABLE_USERS + " (" +
                        COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT, " +
                        COLUMN_USERNAME + " TEXT NOT NULL, " +
                        COLUMN_PASSWORD + " TEXT NOT NULL);";

        public DatabaseHelper(Context context) {
            super(context, DATABASE_NAME, null, DATABASE_VERSION);
        }

        @Override
        public void onCreate(SQLiteDatabase db) {
            db.execSQL(TABLE_CREATE);
        }

        @Override
        public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
            db.execSQL("DROP TABLE IF EXISTS " + TABLE_USERS);
            onCreate(db);
        }

        public boolean registerUser(String username, String password) {
            SQLiteDatabase db = this.getWritableDatabase();
            String insertQuery = "INSERT INTO " + TABLE_USERS +
                    " (" + COLUMN_USERNAME + ", " +
                    COLUMN_PASSWORD + ") VALUES ('" +
                    username + "', '" + password + "');";
            try {
                db.execSQL(insertQuery);
                return true;
            } catch (Exception e) {
                return false;
            }
        }

        public void addTestUser(String username, String password) {
            SQLiteDatabase db = this.getWritableDatabase();
            String insertQuery = "INSERT INTO " + TABLE_USERS +
                    " (" + COLUMN_USERNAME + ", " +
                    COLUMN_PASSWORD + ") VALUES ('" +
                    username + "', '" + password + "');";
            db.execSQL(insertQuery);
        }


        public boolean checkUser(String username, String password) {
            SQLiteDatabase db = this.getReadableDatabase();
            String selectQuery = "SELECT * FROM " + TABLE_USERS +
                    " WHERE " + COLUMN_USERNAME + " = '" + username +
                    "' AND " + COLUMN_PASSWORD + " = '" + password + "';";
            Cursor cursor = db.rawQuery(selectQuery, null);
            if (cursor.moveToFirst()) {
                cursor.close();
                return true;
            } else {
                cursor.close();
                return false;
            }
        }
    }

File: app/src/main/java/com/example/gym_workout/database/DatabaseHelper.kt
package com.example.gym_workout.database

import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class DatabaseHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

    companion object {
        const val DATABASE_NAME = "fitness_tracker.db"
        const val DATABASE_VERSION = 2

        // Table name and columns
        const val TABLE_USERS = "users"
        const val COLUMN_USER_ID = "userId"
        const val COLUMN_FIRST_NAME = "firstName"
        const val COLUMN_LAST_NAME = "lastName"
        const val COLUMN_DATE_OF_BIRTH = "dateOfBirth"
        const val COLUMN_EMAIL = "email"
        const val COLUMN_GENDER = "gender"
        const val COLUMN_HEIGHT = "height"
        const val COLUMN_REASON_FOR_WEIGHT_GAIN = "reasonForWeightGain"
        const val COLUMN_DIET_PLAN = "dietPlan"
        const val COLUMN_WEIGHT = "weight"
        const val COLUMN_GOAL = "goal"
        const val COLUMN_DAILY_WORKING_TIME = "dailyWorkingTime"
        const val COLUMN_FITNESS_LEVEL = "fitnessLevel"
        const val COLUMN_PASSWORD = "password"
        const val COLUMN_STEP_COMPLETED = "stepCompleted"
    }

    override fun onCreate(db: SQLiteDatabase) {
        val createTableQuery = """
            CREATE TABLE $TABLE_USERS (
                $COLUMN_USER_ID TEXT PRIMARY KEY,
                $COLUMN_FIRST_NAME TEXT,
                $COLUMN_LAST_NAME TEXT,
                $COLUMN_DATE_OF_BIRTH TEXT,
                $COLUMN_EMAIL TEXT,
                $COLUMN_GENDER TEXT,
                $COLUMN_HEIGHT INTEGER,
                $COLUMN_REASON_FOR_WEIGHT_GAIN TEXT,
                $COLUMN_DIET_PLAN TEXT,
                $COLUMN_WEIGHT INTEGER,
                $COLUMN_GOAL TEXT,
                $COLUMN_DAILY_WORKING_TIME TEXT,
                $COLUMN_FITNESS_LEVEL TEXT,
                $COLUMN_PASSWORD TEXT,
                $COLUMN_STEP_COMPLETED INTEGER
            )
        """
        db.execSQL(createTableQuery)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        if (oldVersion < 2) {
            db.execSQL("ALTER TABLE $TABLE_USERS ADD COLUMN $COLUMN_EMAIL TEXT")
        }
    }

    /**
     * Save or update user data
     */
    fun saveOrUpdateStepData(data: ContentValues, userId: String): Boolean {
        val db = writableDatabase

        // Check if the user data already exists
        val cursor = db.query(
            TABLE_USERS,
            arrayOf(COLUMN_USER_ID),
            "$COLUMN_USER_ID = ?",
            arrayOf(userId),
            null, null, null
        )

        val rowExists = cursor.moveToFirst()
        cursor.close()

        data.put(COLUMN_USER_ID, userId) // Ensure userId is included in the data

        return if (rowExists) {
            // If data exists, update the existing row
            db.update(TABLE_USERS, data, "$COLUMN_USER_ID = ?", arrayOf(userId)) > 0
        } else {
            // If data does not exist, insert new data
            db.insert(TABLE_USERS, null, data) != -1L
        }.also {
            db.close()
        }
    }


}
